### PR TITLE
KOGITO-7264: Patch PR for KOGITO-7263 with doc improvements in OAuth2 authentication guide

### PR DESCRIPTION
**Pull request by hmanwani-rh**
**JIRA:**https://issues.redhat.com/browse/KOGITO-7264

**Description:**Patch PR for KOGITO-7263 with doc improvements in OAuth2 authentication guide

https://github.com/kiegroup/kogito-docs/pull/29

Please make sure that your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category

### CODE REVIEW

**Review by wmedvede — COMMENTED** [review]
I've left some minor comments, but LGTM.
Thanks

File: serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Orchestration of third-party services using OAuth 2.0 authentication
-====================================================================
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc:2]: Both approaches render the same output. However, now that we've discovered that using heading sections (=, ==) would be better when we'll modularize the docs after DP1.

File: serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc [hunk 2]
```diff
@@ -24,91 +24,70 @@ Orchestration of third-party services using OAuth 2.0 authentication
...
-* {authentication-support-for-openapi-services}[Authentication support for OpenAPI services]
-
-== Introduction
-
-Imagine that you have a set of applications that must resolve currency exchange calculations as part of their regular operations. For that, you need an accurate source of information to get the different exchange rates.
-
-Fortunately, your company has a commercial agreement with Acme Financial Services, and those rates can be queried using their Oauth2 secured services. Additionally, as confidential client, you were granted with proper credentials to access these services as part of the agreement.
-
-However, you don't want to expose Acme's services to the applications.
+This document describes the examples of how you can implement and configure a Serverless Workflow that orchestrates the interaction with an OAuth 2.0 secured REST service.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc:27]: Done

File: serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc [hunk 3]
```diff
@@ -470,109 +430,112 @@ quarkus.oidc-client.acme_financial_oauth.credentials.client-secret.method=basic
...
 
-<3> Disable the Oauth2 server endpoints discovery since the endpoints provided in the <<acme-financial-service-yml, acme-financial-service.yml>> file will be used instead.
-<4> Is the authentication URL of the Oauth2 server.
-<5> Is the relative path or absolute URL of the Oauth2 token endpoint which issues access and refresh tokens.
-<6> Is the client-id to identify the Serverless Workflow against the authorization service, such as `kogito-app`. This identifier must be provided by Acme.
-<7> Is the method to use at the time to send the client-secret for the authentications when the client grant type is used.
-<8> Is the client secret to authenticate the Serverless Workflow against the authorization service when the client grant type is used. This secret must be provided by Acme.
+To configure the link:{quarkus_guide_security_openid_connect_client_url}[Quarkus OpenId Connect Client (OIDC)] for `acme_financial_service_oauth` service, you must follow the rules described in link:{authentication-support-for-openapi-services-oauth-example}[Example of OAuth 2.0 authentication].
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc:442]: Fixed it

File: serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc [hunk 4]
```diff
@@ -24,91 +24,70 @@ Orchestration of third-party services using OAuth 2.0 authentication
...
-[NOTE]
-====
-To get the source code of the example you must clone the {kogito_examples_repository_url}[GitHub repository] and go to the `serverless-workflow-examples/serverless-workflow-oauth2-orchestration-quarkus` directory.
-====
-
-=== Services
+When you use the Acme Financial Services, you can query the exchange rates using their OAuth 2.0 secured services, which you can access using the granted credentials. However, you do not want to expose the services provided by Acme to the applications. In this case, you can implement a Serverless Workflow that resolves the following aspects:
 
-The following services compose the solution:
+* Orchestration with services provided by Acme Financial and currency exchange calculations.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc:40]: Done

File: serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc [hunk 5]
```diff
@@ -24,91 +24,70 @@ Orchestration of third-party services using OAuth 2.0 authentication
...
+* Authentication requirements to access the services provided by Acme Financial.
```
[the remark this answers is not in the document]
Comment (hmanwani-rh) [reply serverlessworkflow/modules/ROOT/pages/security/orchestrating-third-party-services-with-oauth2.adoc:41]: Done

**Pull request merged**